Sources/ForageSDK/Component/ForagePANTextField/ForagePANTextField.swift

```diff
@@ -218,54 +252,72 @@ public class ForagePANTextField: UIView, Identifiable {
 // MARK: - UITextFieldDelegate
 
 extension ForagePANTextField: UITextFieldDelegate {
+    public func textFieldDidBeginEditing(_ textField: UITextField) {
+        delegate?.focusDidChange(self)
+    }
+    
+    public func textFieldDidEndEditing(_ textField: UITextField) {
+        delegate?.focusDidChange(self)
+    }
```

Comment:
My understanding is that the `textFieldDidEndEditing` handler is equivalent to [a blur event](https://developer.apple.com/documentation/uikit/uitextfielddelegate/1619591-textfielddidendediting). @dleis612 was this intentional? I want to make sure that this can't cause GoPuff to misread a blur event as a focus event and cause a wonky experience for users.

Reply:
It is equivalent yes. The event handler is just a general `onFocusDidChange` which encompasses losing focus and gaining focus right?

Reply:
I agree that the name `onFocusDidChange` expresses that it could be called for `blur` and `focus` events. That said, I would think we would want to give SDK users the ability to distinguish between `blur` and `focus` events.

The specific scenario I am concerned about is if GoPuff is doing their own logic to trigger the appearance of the keyboard on focus, then their logic would fire inappropriately if they use our `onFocusDidChange` event subscriber because they'll attempt to launch the keyboard again when the user is trying to leave the input field.

Said differently, I would have thought something like these would offer the information SDK developers would care to know:

**A single method that invokes the same callback but includes a boo to distinguish `focus` from `blur`**
```swift
public func textFieldDidBeginEditing(_ textField: UITextField) {
    delegate?.focusDidChange(self, hasFocus: true) // focus event
}

public func textFieldDidEndEditing(_ textField: UITextField) {
    delegate?.focusDidChange(self, hasFocus: false) // blur event
}
```

**Two distinct methods that invoke either `focus` xor `blur` callback**
```swift
 public func textFieldDidBeginEditing(_ textField: UITextField) {
    delegate?.onFocusEvent(self) // focus event
}

public func textFieldDidEndEditing(_ textField: UITextField) {
    delegate?.onBlurEvent(self) // blur event
}
```

What is your take?

Reply:
For now, the implemented logic should work fine. The event handler returns the state object that has either `isFirstResponder=true` or `isFirstResponder=false`, which they can check to determine if they have focus or not.